core/src/worker/activities/local_activities.rs

```diff
@@ -531,37 +533,92 @@ impl LocalActivityManager {
+
+            let mut resolution = LocalActivityResolution {
+                seq: info.la_info.schedule_cmd.seq,
+                result: status,
+                runtime: info.dispatch_time.elapsed(),
+                attempt: info.attempt,
+                backoff: None,
+                original_schedule_time: info.la_info.schedule_cmd.original_schedule_time,
+            };
+            // We want to generate a cancel task if the reason for failure was a timeout.
+            let task = if is_timeout {
+                Some(ActivityTask {
+                    task_token: task_token.clone().0,
+                    variant: Some(activity_task::Variant::Cancel(Cancel {
+                        reason: ActivityCancelReason::TimedOut as i32,
+                    })),
+                })
+            } else {
+                None
+            };
+
+            match outcome {
+                Outcome::FailurePath { backoff } => {
+                    if let Some(backoff_dur) = backoff {
+                        let fail_or_timeout = if is_timeout { "timed out" } else { "failed" };
                         let will_use_timer =
                             backoff_dur > info.la_info.schedule_cmd.local_retry_threshold;
                         debug!(run_id = %info.la_info.workflow_exec_info.run_id,
                                seq_num = %info.la_info.schedule_cmd.seq,
                                attempt = %info.attempt,
                                will_use_timer,
-                            "Local activity failed, will retry after backing off for {:?}",
-                             backoff_dur
+                            "Local activity {}, will retry after backing off for {:?}",
+                            fail_or_timeout,
+                            backoff_dur
                         );
                         if will_use_timer {
-                            // We want this to be reported, as the workflow will mark this
-                            // failure down, then start a timer for backoff.
-                            return LACompleteAction::LangDoesTimerBackoff(
-                                backoff_dur.try_into().expect("backoff fits into proto"),
-                                info,
-                            );
+                            // This la needs to write a failure marker, and then we will tell lang how
+                            // long of a timer to schedule to back off for. We do this because there are
+                            // no other situations where core generates "internal" commands so it is
+                            // much simpler for lang to reply with the timer / next LA command than to
+                            // do it internally. Plus, this backoff hack we'd like to eliminate
+                            // eventually.
+                            resolution.backoff =
+                                Some(backoff_dur.try_into().expect("backoff fits into proto"));
```

Comment:
When would try_into fail?

Reply:
It won't, you'd have to have a backoff of thousands of years. Proto internally uses i64 and Duration uses u64

Reply:
Yeah, probably truncate down is the way to go.